src/opm/io/eclipse/rst/state.cpp

```diff
@@ -209,23 +209,30 @@ void RstState::add_udqs(const std::vector<int>& iudq,
         auto& udq = this->udqs.back();
         if (udq.var_type == UDQVarType::WELL_VAR) {
             for (std::size_t well_index = 0; well_index < this->wells.size(); well_index++) {
-                auto well_value = dudw[ udq_index * this->header.num_wells + well_index];
+                auto well_value = dudw[ udq_index * this->header.max_wells_in_field + well_index];
+                if (well_value == UDQ::restart_default)
+                    continue;
+
                 const auto& well_name = this->wells[well_index].name;
-                udq.add_well_value( well_name, well_value );
+                udq.add_value(well_name, well_value);
             }
         }
 
         if (udq.var_type == UDQVarType::GROUP_VAR) {
             for (std::size_t group_index = 0; group_index < this->groups.size(); group_index++) {
                 auto group_value = dudg[ udq_index * this->header.ngroup + group_index];
```

Comment:
Is this really supposed to say `header.ngroup`?  I would have thought `header.max_groups_in_field` would be more appropriate since the window size is `INTEHEAD[NGMAXZ]`.

Reply:
You are probably right

Reply:
> You are probably right

Okay, then I'll merge this and fix that issue in #2556 too.